src/concepts/layout/README.md

```diff
@@ -0,0 +1,216 @@
+# Layout
+
+The coordinate system in Ratatui runs left to right, top to bottom, with the origin `(0, 0)` in the
+top left corner of the terminal. The x and y coordinates are represented by u16 values and are
+generally listed in that order in most places.
+
+```svgbob
+( 0, 0 )  -----------> x (columns)
+
+   |
+   |
+   |
+   |
+   v
+
+   y (rows)
+```
+
+Layouts and widgets form the basis of the UI in Ratatui. Layouts dictate the structure of the
+interface, dividing the screen into various sections using constraints, while widgets fill these
+sections with content.
+
+When rendering widgets to the screen, you first need to define the area where the widget will be
+displayed. This area is represented by a rectangle with a specific height and width in the buffer.
+You can specify this rectangle as an absolute position and size, or you can use the [`Layout`]
+struct to divide the terminal window dynamically based on constraints such as Length, Min, Max,
+Ratio, Percentage.
+
+The following example renders "Hello world!" in the center of the screen, by manually calculating
+the rect to render within.
+
+```rust
+let greeting = "Hello world!";
+let full_screen = frame.size();
+let area = Rect {
+    x: (full_screen.x - greeting.len() as u16) / 2,
+    y: full_screen.size().y / 2,
+    width: full_screen.width,
+    height: full_screen.height,
+};
+frame.render_widget(Paragraph("Hello world!"), area);
+```
+
```

Comment:
I would add a `admonish tip` here saying something like "if you actually wanted to do this, you should use `Alignment::Center`" and show a one line example of that.

Reply:
I'd prefer not to put a worse example when this can be achieved another way simply, I'll swap it out for a simple for loop.